GTScript-Parallel-model.md

```diff
@@ -296,3 +296,118 @@ with computation(...):
+
+This code is translated in pseudo-code as:
+
+```python
+for k in range(start, end):
+    a[i:I,j:J,k-1] = c[i:I,j:J,k]
+    b[i:I,j:J,k+1] = 2 * a[i:I,j:J,k] # b contains 2 times the previous values in a before the assignment of c, since a and b are evaluated at the same k-level
+
+for k in range(end, start):
+    a[i:I,j:J,k-1] = d[i:I,j:J,k]
+    b[i:I,j:J,k+1] = 2 * a[i:I,j:J,k] # b is going to contain the 2*d except for the firt iteration point, max2.
+
+for k in shuffle(range(start, end)):
+    a[i:I,j:J,k-1] = c[i:I,j:J,k]
+    b[i:I,j:J,k+1] = 2 * a[i:I,j:J,k] # depending on the order in which k-loop is executed we can find values from a or from c
+```
+
+### Another code sample with conditionals:
+
+```python
+with computation(FORWARD):  # Forward computation
+    with interval(start, end):
+        if (x < 0):
+            a[0,0,-1] = 1
+        else
+            a[0,0,1] = 2
+
+        b = a
+```
+
+Which translates to
+
+```python
+for k in range(start, end):
+    mask[i:I, j:J] = x < 0
+    parfor ij:
+        if (mask):
+            a[0,0,-1] = 1
+        else
+            a[0,0,1] = 2 # This will be visible in the same computation
+
+    parfor ij:
+        b = a
+```
+In this case `b` will contains 2 where `x<0`, otherwise the values contained in `a` before the computation started. This is true except for the first iteration point `min1`, for which `b` contains just the previous values contained in `a`.
+
+
+### Another code sample with conditionals and offset reads:
+
+```python
+with computation(FORWARD):  # Forward computation
+    with interval(start, end):
+        if (x < 0):
+            a[0,0,-1] = 1
+        else
+            a[0,0,1] = 2
+
+        b = a[0,0,-1]
+```
+
+Which translates to
+
+```python
+for k in range(min1, max1): #min < max
+    mask[i:I, j:J] = x < 0
+    parfor ij:
+        if (mask):
+            a[0,0,-1] = 1 # This will be visible in the same computation in assignment of b later
+        else
+            a[0,0,1] = 2 # This will be visible, eventually, in two k iterations
+
+    parfor ij:
+        b = a[0,0,-1]
+```
+
+</details>
+
+There are different uses in which writing off-center can work with clear semantics: write at a location before the iteration loop reaches it (e.g., offset greater than zero in a FORWARD loop), and write at a location after the iteration loop reaches it (e.g., offset less than zero in a FORWARD loop). Both cases are valid. In the first we want to make the values visible in the same computation, in the second case we want to make the values available for subsequent computations. There are cases with conditionals in which the two cases above can be mixed up, but they are deterministic and creates no problems, even though understanding the code by the programmers can be difficult.
+
+### Policy for off-center writes
+
+The policy is the following:
+
+- If iteration policy is PARALLEL then write off-center in the vertical direction is forbidden and raise a compile error
```

Comment:
- There is no `PARALLEL`
- I see you have here the explanation for the example above. Maybe move the policy first?